### PR TITLE
[CF-429] Fix cinder migration using non-root user

### PR DESCRIPTION
**Pull request by antonf**
When doing command piping like this: 'dd of=in_file | ssh user@host dd of=out_file', it will not work if user don't have access to this file. Also, we can't use sudo since it's in different cloud and password can differ from password on source cloud. So the only way to implement this without terrible hacks is through nc.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
Please add a description

File: cloudferry/lib/os/storage/plugins/copy_mechanisms.py [hunk 1]
```diff
@@ -82,27 +83,47 @@ def copy(self, context, source_object, destination_object):
...
-            rr.run(copy.format(src_file=source_object.path,
-                               dst_user=dst_user,
-                               dst_host=dst_host,
-                               ssh_opts=ssh_opts,
-                               dst_device=destination_object.path,
-                               progress_view=progress_view))
+            # First step: prepare netcat listening on aux_port on dst and
+            # forwarding all the data to block device
+            dump_cmd = ('/bin/bash -c \'nc -l -p {aux_port} -s 127.0.0.1 | '
+                        'dd of={dst_device}\' &')
```
Comment (roman-verchikov) [inline cloudferry/lib/os/storage/plugins/copy_mechanisms.py]: Am I reading this line correctly: `nc` will be spawned for each migrated volume in the background and will never be killed? The `&` sign in the end confuses me, and also I don't see any termination for the spawned `nc` process. If I'm reading it correctly, this needs to be resolved, we should not leave `nc` processes running in the background.

**Comment by roman-verchikov** [comment]
Please take a look at comment inline and add description to pull request and commit message

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by antonf** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by antonf** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Merge check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3598/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.storage.plugins.copy_mechanisms
cloudferry/lib/os/storage/plugins/copy_mechanisms.py:18: [W0611(unused-import), ] 
Unused import re

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by yosharma** [comment]
+1, Tested on staging environment. Fix is working fine

**Comment by roman-verchikov** [comment]
+1

**Comment by roman-verchikov** [comment]
those `sleep`s are awesome :)

**Comment by SVilgelm** [comment]
👎 Looks good to me

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**